### PR TITLE
Add some Block Editor Tracking E2E tests

### PR DESCRIPTION
**Pull request by david-szabo97**
#### Changes proposed in this Pull Request

* E2E tests for https://github.com/Automattic/wp-calypso/pull/53405
* E2E tests for https://github.com/Automattic/wp-calypso/pull/53501 (template parts only)

#### Testing instructions

```bash
cd test/e2e

# Page editor
./node_modules/.bin/mocha specs/specs-wpcom/wp-calypso-gutenberg-page-editor-tracking-spec.js
# Post Editor
./node_modules/.bin/mocha specs/specs-wpcom/wp-calypso-gutenberg-post-editor-tracking-spec.js
# Site Editor
./node_modules/.bin/mocha specs/specs-wpcom/wp-calypso-gutenberg-site-editor-tracking-spec.js
```

Related to #53410

### CODE REVIEW

File: test/e2e/lib/components/site-editor-component.js [hunk 1]
```diff
@@ -120,4 +120,33 @@ export default class SiteEditorComponent extends AsyncBaseContainer {
 			return await this.driver.findElement( insertedBlockLocator ).getAttribute( 'id' );
 		} );
 	}
+
+	async removeBlock( blockID ) {
```
Comment (Addison-Stavlo) [inline test/e2e/lib/components/site-editor-component.js:131]: If we aren't interested in testing the interaction with the block toolbar here, I wonder if we should be utilizing the block-editor store action `removeBlock` instead?

I wonder how robust this flow with selectors will actually be.  Sometimes Ive had issues where the block toolbar doesn't show up until I scroll a little, minor issues like that could be problematic here.  If the 'remove block' button is no longer last-of-type in some circumstances, etc.

Similarly if selectors change or classes like `.components-menu-group` end up elsewhere in the editor we could be clicking the wrong thing.  I think currently the ellipsis/options dropdown from the top-right of the header also uses this menu group component, but currently it closes as soon as the toolbar menu is clicked so there is currently no conflict there.

File: test/e2e/lib/gutenberg/tracking/utils.js [hunk 2]
```diff
@@ -12,10 +12,10 @@ export function getTotalEventsFiredForBlock( eventsStack, event, block ) {
 	return getEventsFiredForBlock( eventsStack, event, block ).length;
 }
 
-export async function clearEventsStack() {
+export async function clearEventsStack( driver ) {
 	// Reset e2e tests events stack after each step in order
 	// that we have a test specific stack to assert against.
-	await this.driver.executeScript( `window._e2eEventsStack = [];` );
```
Comment (Addison-Stavlo) [inline test/e2e/lib/gutenberg/tracking/utils.js:18]: was this not working? 😆  pun slightly intended.

**Review by Addison-Stavlo — COMMENTED** [review]
This is looking pretty good to me.  Just a minor Q about creating `removeBlock` via selectors and timeouts vs. using the block editor store.

File: test/e2e/lib/gutenberg/tracking/utils.js (hunk 2 above)
Comment (david-szabo97) [reply test/e2e/lib/gutenberg/tracking/utils.js:18]: It took me a while to figure it out 😆

File: test/e2e/lib/components/site-editor-component.js (hunk 1 above)
Comment (david-szabo97) [reply test/e2e/lib/components/site-editor-component.js:131]: I don't think we should prefer store actions over actual interface clicking.

Though we are actually tracking these by intercepting store actions, I'd still prefer these tests to break when the UI changes. That might indicate that we should take a look at the test and make sure it's still working properly. (If the UI changes something else might change too that might be related)

`removeBlock` is a copy-paste from Gutenberg Editor Component in this case. With a few tweaks to make it work with site editor iframing.

Comment (Addison-Stavlo) [reply test/e2e/lib/components/site-editor-component.js:131]: makes sense to me!

**Review by Addison-Stavlo — APPROVED** [review]

File: test/e2e/lib/components/site-editor-component.js [hunk 3]
```diff
@@ -127,4 +127,33 @@ export default class SiteEditorComponent extends AsyncBaseContainer {
...
+		await driverHelper.clickWhenClickable(
+			this.driver,
+			By.css( '.block-editor-block-settings-menu' )
+		);
+		await driverHelper.waitUntilElementLocatedAndVisible(
+			this.driver,
+			By.css( '.components-menu-group' ),
+			this.explicitWaitMS / 5
+		);
+		await this.driver.sleep( 1000 );
```
Comment (WunderBart) [inline test/e2e/lib/components/site-editor-component.js:151]: Why the extra sleep here?

Comment (david-szabo97) [reply test/e2e/lib/components/site-editor-component.js:151]: This is a copy-paste from `GutenbergEditorComponent#removeBlock`, but adapted to the site editor. The `sleep` is not documented there either, so I'm not quite sure why it's required.

Comment (WunderBart) [reply test/e2e/lib/components/site-editor-component.js:151]: Ah, OK. Anyway, a good approach is to verify if those mysterious `sleeps` are actually necessary whenever bumped into one, especially when copypasting from somewhere else. This way we're not adding any unnecessary time to our E2Es.

**Pull request merged**